parfive/downloader.py

```diff
@@ -535,13 +571,30 @@ async def _get_http(
                 )
                 if resp.status < 200 or resp.status >= 300:
                     raise FailedDownload(filepath_partial, url, resp)
-                filepath, skip = get_filepath(filepath_partial(resp, url), overwrite)
-                if skip:
-                    parfive.log.debug(
-                        "File %s already exists and overwrite is False; skipping download.",
-                        filepath,
-                    )
-                    return url, str(filepath)
+                filepath, file_exists = get_filepath(filepath_partial(resp, url), overwrite)
+                # Get the expected checksum from the headers
+                if checksum is True:
+                    checksum = resp.headers.get("Repr-Digest", resp.headers.get("Content-Digest", None))
+                    if checksum is None:
+                        parfive.log.info(
+                            "Expected server to provide checksum for url '%s' but none returned.", url
+                        )
+                if file_exists:
+                    if isinstance(checksum, str):
```

Comment:
At this point is checksum either a string or ``None``? In which case you could just do ``if checksum``?

Reply:
ha, doing `if checksum` made mypy angry so this was the easiest "fix" but yes lol